src/Notepads/Core/NotepadsCore.cs

```diff
@@ -798,15 +799,46 @@ private async void Sets_SetDraggedOutside(object sender, SetDraggedOutsideEventA
         {
             if (Sets.Items?.Count > 1 && e.Set.Content is ITextEditor textEditor)
             {
-                // Only allow untitled empty document to be dragged outside for now
-                if (!textEditor.IsModified && textEditor.EditingFile == null)
+                if (textEditor.FileModificationState != FileModificationState.RenamedMovedOrDeleted)
                 {
                     DeleteTextEditor(textEditor);
-                    await NotepadsProtocolService.LaunchProtocolAsync(NotepadsOperationProtocol.OpenNewInstance);
+                    var message = new ValueSet();
+                    message.Add("EditorData", JsonConvert.SerializeObject(await BuildTextEditorSessionData(textEditor), Formatting.Indented));
+                    message.Add("LastSavedText", (textEditor.IsModified || textEditor.FileModificationState == FileModificationState.Modified) && textEditor.EditingFile != null ? textEditor.LastSavedSnapshot.Content : null);
+                    message.Add("PendingText", textEditor.IsModified ? textEditor.GetText() : null);
```

Comment:
Did you verify the upper size limit of message when transferring using the launch protocol? Try test with a large file to see if it works.

Reply:
> Did you verify the upper size limit of message when transferring using the launch protocol? Try test with a large file to see if it works.

This is another reason I did not implement the dragging for non empty eidtors.

Reply:
Made the approach similar to session snapshot, now it will work.